### DISCUSSION TITLE
Are you a NATS End-User?

### DISCUSSION THREAD

**Discussion by gcolliso** [nats-io/nats-general#58] [Show and tell]
## 👋 Welcome!
  # Are you using NATS.io?

If you are reviewing or already using NATS, thank you!  

It is users like you in the OSS community and ecosystem that help make projects like NATS successful.  We love to hear from our users, learn about how you are using NATS, and discover what we could add or improve to make NATS better for you!

## Interested in sharing your experience?

If you would like to share here, we'd love to see a comment answering any of the following questions.

```text
Company/Project Name:

Website:

How did you hear about NATS?

How are you using NATS?  

Do you have any feature requests?

What would you change about NATS?

Are you running NATS in production?

Any other comments?
```

Alternatively, feel free to email us at info@nats.io, or join our [slack organization](https://slack.nats.io).

To request new features or enhancements don’t hesitate to add a request at [nats-general/issues](https://github.com/nats-io/nats-general/issues/new/choose) or any of our other repositories.

**Comment by gcolliso** [comment]
**Company name:**  [Synadia](https://synadia.com) 

**How did you hear about NATS?**
Our CEO, Derek Collison created NATS.

**What are you using NATS for?**
To provide secure global connectivity for applications, devices, and endpoints and to democratize access to streams and services across the world.

**Do you have any feature requests?**
Something around schema management would be great!

**What would you change about NATS?**
As we come up with ideas we’ll open issues or PRs.

**Are you running NATS in production?**
Yes, for three years now.  We are happy to be a part of the NATS community and contribute to the NATS.io project.


<img src="https://user-images.githubusercontent.com/5532951/122467987-c272d080-cf80-11eb-977a-411572c4be45.png" height="25%" width="25%">

**Comment by klynton** [comment]
**Company/Project Name:** [Opalstack](https://www.opalstack.com)

**Website:** https://www.opalstack.com

**How did you hear about NATS?**

I saw Derek Collison give a presentation on it a few years ago.

**How are you using NATS?**  

We're using it to trigger downloads of configuration data from Minio and as an event bus.

**Do you have any feature requests?**

I'd love some more opinionated operational documentation. The documentation is good but it's very general. A series of "do this, not that" tutorials for operationalizing production services would be great.

**What would you change about NATS?**

I think a lot of the JetStream work will fix most of the things I'd change.

**Are you running NATS in production?**

Yes, for just over 2 years.

**Any other comments?**

NATS is one of the most hands-off parts of our stack. It stays online for months at at time (between releases), delivers millions of messages to hundreds of hosts, and does it all in less than 50MB of RAM.

**Reply by gcolliso** [reply]
@LLLLimbo Thank you for using NATS! Would it be possible for you to move your entry to a separate comment rather than a reply to Opalstack's entry?

**Comment by LLLLimbo** [comment]
Company name: Seeiner

How did you hear about NATS?
Found on ThoughtWorks technology radar

What are you using NATS for?
Provide support for communication between microservices, and device communication between the cloud and the edge.

Are you running NATS in production?
Not yet, but soon.

**Comment by cbrake** [comment]
Company/Project Name:

Website:

> How did you hear about NATS?

https://changelog.com/gotime/130

> How are you using NATS?  

We use NATS in https://github.com/simpleiot/simpleiot. 

> Do you have any feature requests?

We current embed NATS for easy deployment to edge devices, and it would be neat to explore a lightweight NATS for edge devices with only the core pub/sub stuff. 

Programmatically managing authn/authz would also be useful.

The above might already be possible -- something we need to look into, but have not had time yet.

> What would you change about NATS?

Can't think of anything other than the above.

> Are you running NATS in production?

Yes, several systems based on SIOT are in production.

> Any other comments?

Great project -- enjoy using it. Appreciate the focus on simplicity, single pure Go app for easy deployment, and able to embed when needed.

**Reply by gcolliso** [reply]
Thank you @cbrake

**Comment by mcosta74** [comment]
**Company/Project Name**: FileWave

**Website**: https://www.filewave.com/

**How did you hear about NATS?**

We were looking for an high-performance messaging system and we found NATS (from CNCF landscape and other resources/articles)

**How are you using NATS?**

NATS is now the key component of our messaging system; we use it for communication between our services and user interfaces.

**Do you have any feature requests?**

Would be great to have a better way to authenticate remote users (at the moment you need to provide credential files that is not nice if the user needs to authenticate using a web application)

**What would you change about NATS?**

We are using client libraries in C, Python, Javascript (WS) and Go; would be nice if these libraries could be in sync with the server release (able to use the latest server features)

**Are you running NATS in production?**

Yes, since `v14.8.0` of our product

**Any other comments?**

Thanks a lot for the great job

**Reply by gcolliso** [reply]
Thank you for the feedback and for using NATS @mcosta74 ! 😃

**Comment by tachunwu** [comment]
Company/Project Name: My personal master research

Website: check my github

How did you hear about NATS?
via CNCF Landscape

How are you using NATS?  
for sharding, workflow management (integrate with 3rd part API), distributed transaction( Google Spanner like 2PC & Slog/Calvin like deterministic transaction)

Do you have any feature requests?
yes NATS for sharding cluster, distributed transaction, SQL-like materialize (streaming database),distributed cache plug-ins

What would you change about NATS?
for large scale distributed 

Are you running NATS in production?
yes

Any other comments?
thx so much that NATS speedup my research

**Reply by gcolliso** [reply]
Thank you for sharing your feedback [@tachunwu](https://github.com/tachunwu)!

**Comment by pablitovicente** [comment]
**Company/Project Name**

Personal Projects/Evaluating for using it in a future product.

**Website**

None yet.

**How did you hear about NATS?**

Some years ago I was looking for alternatives to MQTT, Redis, and RabbitMQ and I was amazed by the simplicity and performance of NATS. Jetstream made it a clear winner on many use cases against those other three brokers.

**How are you using NATS?**

Using it as the main event broker for a micro service architecture that needs to be fast and easy to maintain.

**Do you have any feature requests?**

Some more documentation on streams/consumers could lower the learning curve a bit. Better documentation on how to balance subjects/streams and subject mapping would be great.

- There’s an open issue to support MQTT 5 if that happens NATS could be the only broker you ever need
- Flexible external authentication API

**What would you change about NATS?**

Hard to tell so far it seems an extremely solid solution. It does need more visibility in the industry as it is not always well know yet.

**Are you running NATS in production?**

Not yet, but I hope to do so in 2023

**Any other comments?**

Thanks for making NATS and huge kudos to the team that always responds and actively supports us!

**Reply by gcolliso** [reply]
Thank you for your feedback @pablitovicente !

**Comment by nithishepic** [comment]
**Company/Project Name**: Epicode Private Limited

**Website**: https://www.epicode.in

**How did you hear about NATS?**
We were exploring options for high performance, highly available messaging frameworks and came across NATS in CNCF landscape.

**How are you using NATS?**
NATS is now the heart of our solutions. Our applications use our own in memory DB to store transient information and we use NATS to sync changes to the DB across multiple instances of our applications. We expose our CPaaS APIs as NATS requests and our partners subscribe to our NATS subjects to get call/chat events in real time.

**Do you have any feature requests?**
None at the moment.

**What would you change about NATS?**
Nothing at the moment.

**Are you running NATS in production?**
Yes

**Any other comments?**
Excellent project. Appreciate the high quality NATS clients in multiple programming languages.

**Reply by bruth** [reply]
@nithishepic We really appreciate this! Very cool use case and glad it has been impactful in your company!

**Comment by tachunwu** [comment]
Company/Project Name: Awareness Labs/Rainforest

Website: https://awareness-labs.github.io/

How did you hear about NATS?
We were researching high-performance messaging systems (in Go) suitable for our distributed data architecture. NATS emerged as a leading solution in this space.

How are you using NATS?
Rainforest uses NATS as its core messaging system. It's at the heart of our platform, enabling lightweight, secure, and scalable real-time communication. NATS ensures efficient data synchronization and inter-service communication within our distributed architecture.

Do you have any feature requests?
As of now, NATS sufficiently addresses our needs. However, we'd always welcome improvements in areas of distributed tracing and monitoring capabilities, and Leaf Nodes is our favorite so add more stable into Leaf Nodes！

What would you change about NATS?
At this juncture, NATS aligns well with our requirements. However, better documentation or more in-depth examples tailored for distributed data mesh architectures would be appreciated.

Are you running NATS in production?
Yes, NATS is a critical component in our production environment, ensuring efficient real-time communication in Rainforest Data Mesh core engine.

Any other comments?
We believe that NATS' lightweight and high-performance characteristics play a pivotal role in Rainforest ability to provide a nimble and scalable data mesh model. We look forward to seeing NATS continue to evolve and cater to the needs of contemporary data infrastructure projects like ours.

**Reply by gcolliso** [reply]
Thank you @tachunwu ❤️

**Reply by bruth** [reply]
@tachunwu Would you be interested in writing a blog post for NATS.io? This is a really interesting project and I would be happy to facilitate.

**Reply by Spencerx** [reply]
Would it be possible to make it practical option for architects, admins or devs who might have to wear all those hats when they have to use message broker and queue software to have some consistent grace in load balancing.

**Comment by tamalsaha** [comment]
Company/Project Name: AppsCode Inc.

Website: https://appscode.com

How did you hear about NATS? KubeCon, Twitter

How are you using NATS? We are self-hosting NATS on Kubernetes for our billing infra.

Do you have any feature requests? NATS.io 1.0 helm charts is insanely complex. So, we stick to the previous helm charts and change the docker image tags to use latest nats-server. I think it will help if that helm chart can be either reversed to old style or have better documentation on how to do things in the new chart.

What would you change about NATS? More stability, resiliency on the Jetstream cluster setup

Are you running NATS in production? Yes.

Any other comments? No.

**Reply by gcolliso** [reply]
Thank you @tamalsaha

**Comment by Toyz** [comment]
Company/Project Name: 
**protoc-gen-nats-micro**

Website:
[protoc-gen-nats-micro
](https://github.com/Toyz/protoc-gen-nats-micro)

How did you hear about NATS?
Mostly google when looking for a Pub/Sub

How are you using NATS? 
**Service Discovery for multiple microservices**

Do you have any feature requests? 
**No**

What would you change about NATS?
**Some of the errors jetstream can throw are confusing including when using `nack-box`**

Are you running NATS in production?
**Yes with over 100 handlers sing `NATS.micro`**

Any other comments? No

**Comment by heyjoja** [comment]
Company/Project Name:
Astrolle

Website:
astrolle.com

How did you hear about NATS?
Mostly through research while looking for a reliable Pub/Sub and event-driven communication system for our microservices architecture.

How are you using NATS?
We use NATS as part of Astrolle’s event-driven architecture to connect multiple microservices, coordinate internal processes, and support scalable communication between different platform modules.

Do you have any feature requests?
Not at the moment.

What would you change about NATS?
Some JetStream errors can be difficult to understand, especially when debugging message retries, acknowledgements, or stream configuration issues. Or maybe we need more courses :) 

Are you running NATS in production?
Yes. We use NATS in production as part of our microservices ecosystem to support real-time communication, service orchestration, and scalable backend operations.

Any other comments?
NATS has helped us build a more flexible and scalable architecture for Astrolle as we continue expanding our platform.


<br>
<br>
<br>
<br>


![Astrolle Logo](https://astrolle.com/assets/imgs/logo/logo-astrolle.webp)

**Reply by gcolliso** [reply]
Thank you @heyjoja for sharing this information about Astrolle and using NATS!
